### PR TITLE
Adding Entities, Components for Backend

### PR DESCRIPTION
**Pull request by MrVintage710**
PR for #19

This PR will add the following:
- **Entities**. These are the base of our data system.
- **Components**. These are the parts of an entity that define what it is.
- **Archetypes**. This a describer that defines a set of components.

### CODE REVIEW

**Comment by Gearhartlove** [comment]
:eyes: I'll take a look at this tomorrow, looks super promising!

**Comment by Gearhartlove** [comment]
Last thing to note is that the 3 tests `src-types/src/entity.rs` pass perfectly. The doc tests however, above `impl Entity` in`fn add ...`, `fn get ...` , and `fn remove ...` do not work currently. When I run `cargo test` the main 3 tests fail and the other doc tests fail. 

This is not a major issue, but it is confusing when you run a `cargo test`. To fix this we need to 
- import correct dependencies in each doc test (I know repetitive :sweat_smile:)
- create structs for `A` and `B` for each doc test

**Review by Gearhartlove — CHANGES_REQUESTED** [review]
Here is my review! yay!

File: src-types/src/entity.rs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+use crate::component::{Archetype, Component, ComponentGroup};
+
+//=========================================================================================================================
+//           Entity Struct
+//=========================================================================================================================
+
+/// An object that holds up to one of each Component type. This is how everything is represented in vitruvian.
+#[derive(Default)]
+pub struct Entity {
+    compnents : HashMap<TypeId, RefCell<Box<dyn Component>>>,
```
[the remark this answers is not in the document]
Comment (MrVintage710) [reply src-types/src/entity.rs]: This naming was fixed!

File: src-tauri/src/lib.rs [hunk 2]
```diff
@@ -13,4 +15,13 @@ pub fn run() {
         .invoke_handler(tauri::generate_handler![greet])
         .run(tauri::generate_context!())
         .expect("error while running tauri application");
+
+    let bulk = vitruvian_types::prelude::Bulk(10);
+    println!("{}", bulk.to_json().as_str().unwrap())
```
[the remark this answers is not in the document]
Comment (MrVintage710) [reply src-tauri/src/lib.rs]: I removed this as it was test code anyways. Problem solved!

File: src-tauri/Cargo.lock
[the remark this answers is not in the document]
Comment (MrVintage710) [reply src-tauri/Cargo.lock:1]: Yup! Its should be good from here on out

[the remark this answers is not in the document]
Comment (Gearhartlove) [reply src-tauri/Cargo.lock:1]: We are git ignoring the Cargo.lock so we won't see this issue anymore :+1:

**Review by Gearhartlove — APPROVED** [review]

**Pull request merged**